.github/scripts/trymerge.py

```diff
@@ -733,11 +739,28 @@ def gen_commit_message(self, filter_ghstack: bool = False) -> str:
         msg += f"Approved by: {approved_by_urls}\n"
         return msg
 
-    def merge_into(self, repo: GitRepo, *, force: bool = False, dry_run: bool = False, comment_id: Optional[int] = None) -> None:
+    def merge_into(self, repo: GitRepo, *,
+                   force: bool = False,
+                   dry_run: bool = False,
+                   comment_id: Optional[int] = None) -> None:
         # Raises exception if matching rule is not found
         find_matching_merge_rule(self, repo, force=force, skip_internal_checks=can_skip_internal_checks(self, comment_id))
-        if repo.current_branch() != self.default_branch():
-            repo.checkout(self.default_branch())
+        self.merge_changes(repo, force, comment_id)
```

Comment:
should this have a dry_run param?

Reply:
eh i guess it is the same regardless

Reply:
the dry run only really matters for push.